sdk/iot/pnp/tests/cmocka/test_az_iot_pnp_client_telemetry.c

```diff
@@ -0,0 +1,367 @@
+          &client, test_component_name, mqtt_topic, NULL, &mqtt_topic),
+      AZ_OK);
+
+  az_span_for_test_verify(
+      mqtt_topic,
+      g_test_correct_pnp_topic_content_encoding,
+      _az_COUNTOF(g_test_correct_pnp_topic_content_encoding) - 1,
+      TEST_SPAN_BUFFER_SIZE);
+}
+
+static void test_az_iot_pnp_client_telemetry_publish_topic_get_content_type_and_encoding_succeed(
+    void** state)
+{
+  (void)state;
+
+  az_iot_pnp_client_options options = az_iot_pnp_client_options_default();
+  options.content_type = test_content_type;
+  options.content_encoding = test_content_encoding;
+
+  az_iot_pnp_client client;
+  assert_int_equal(
+      az_iot_pnp_client_init(
+          &client, test_device_hostname, test_device_id, test_root_interface_name, &options),
+      AZ_OK);
+
+  uint8_t mqtt_topic_buf[TEST_SPAN_BUFFER_SIZE];
+  az_span mqtt_topic = az_span_for_test_init(mqtt_topic_buf, 0, _az_COUNTOF(mqtt_topic_buf));
+
+  assert_int_equal(
+      az_iot_pnp_client_telemetry_get_publish_topic(
+          &client, test_component_name, mqtt_topic, NULL, &mqtt_topic),
+      AZ_OK);
+
+  az_span_for_test_verify(
+      mqtt_topic,
+      g_test_correct_pnp_topic_content_type_and_encoding,
+      _az_COUNTOF(g_test_correct_pnp_topic_content_type_and_encoding) - 1,
+      TEST_SPAN_BUFFER_SIZE);
+}
+
+static void test_az_iot_pnp_client_telemetry_publish_topic_get_with_small_buffer_fails(void** state)
+{
+  (void)state;
+
+  az_iot_pnp_client client;
+  assert_int_equal(
+      az_iot_pnp_client_init(
+          &client, test_device_hostname, test_device_id, test_root_interface_name, NULL),
+      AZ_OK);
+
+  uint8_t mqtt_topic_buf[_az_COUNTOF(g_test_correct_pnp_topic_no_options) - 2];
+  az_span mqtt_topic = az_span_for_test_init(mqtt_topic_buf, 0, _az_COUNTOF(mqtt_topic_buf));
+
+  assert_int_equal(
+      az_iot_pnp_client_telemetry_get_publish_topic(
+          &client, test_component_name, mqtt_topic, NULL, &mqtt_topic),
+      AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY);
+}
+
+static void test_az_iot_pnp_client_telemetry_publish_topic_get_content_type_with_small_buffer_fails(
+    void** state)
+{
+  (void)state;
+
+  az_iot_pnp_client_options options = az_iot_pnp_client_options_default();
+  options.content_type = test_content_type;
+
+  az_iot_pnp_client client;
+  assert_int_equal(
+      az_iot_pnp_client_init(
+          &client, test_device_hostname, test_device_id, test_root_interface_name, &options),
+      AZ_OK);
+
+  uint8_t mqtt_topic_buf[_az_COUNTOF(g_test_correct_pnp_topic_content_type) - 2];
+  az_span mqtt_topic = az_span_for_test_init(mqtt_topic_buf, 0, _az_COUNTOF(mqtt_topic_buf));
+
+  assert_int_equal(
+      az_iot_pnp_client_telemetry_get_publish_topic(
+          &client, test_component_name, mqtt_topic, NULL, &mqtt_topic),
+      AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY);
+}
+
+static void
+test_az_iot_pnp_client_telemetry_publish_topic_get_content_encoding_with_small_buffer_fails(
+    void** state)
```

Comment:
just as a note: going forward each function doesn't need this `void** state`. Leaving it empty is valid for a variadic parameter. So future tests can just be `..._fails()` and can remove the `(void)state`. For now you can leave it because I will issue a PR removing all of them with regex.